lib/matplotlib/legend.py

```diff
@@ -74,10 +74,12 @@ def artist_picker(self, legend, evt):
         return self.legend.contains(evt)
 
     def finalize_offset(self):
-        update_method = cbook._check_getitem(
-            {"loc": self._update_loc, "bbox": self._bbox_to_anchor},
+        updater_name = cbook._check_getitem(
+            # If e.g. _update == "loc", self._bbox_to_anchor may not exist, so
+            # this dict can't directly map to attributes.
+            {"loc": "_update_loc", "bbox": "_bbox_to_anchor"},
             update=self._update)
-        update_method(self.get_loc_in_canvas())
+        getattr(self, updater_name)(self.get_loc_in_canvas())
```

Comment:
This is too much unnecessary indirection. I'd rather go with:
~~~
if self._update == 'loc':
    self._update_loc(self.get_loc_in_canvas())
elif self._update == 'bbox':
    self._bbox_to_anchor(self.get_loc_in_canvas())
~~~

`self._update` is guaranteed to be one of the two (see `DraggableLegend.__init__`). No checks necessary.

Reply:
fair point, edited.